### PR TITLE
DBZ-6306 Use connection string configuration in ConnectorConfiguration#forMongoDbContainer

### PR DESCRIPTION
**Pull request by ani-sha**
https://issues.redhat.com/browse/DBZ-6306

### CODE REVIEW

**Review by Naros — APPROVED** [review]
These seem okay to me, but I'll defer to Jakub as a final say.

File: debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/ConnectorConfiguration.java [hunk 1]
```diff
@@ -68,14 +66,13 @@ public static ConnectorConfiguration forJdbcContainer(JdbcDatabaseContainer<?> j
...
+    public static ConnectorConfiguration forMongoDbContainer(MongoDbContainer mongoDbContainer) {
         final List<Integer> exposedPorts = mongoDbContainer.getExposedPorts();
 
         ConnectorConfiguration configuration = new ConnectorConfiguration();
         configuration.with(CONNECTOR, "io.debezium.connector.mongodb.MongoDbConnector")
-                .with(HOSTS, "rs0/" + mongoDbContainer.getContainerInfo().getConfig().getHostName()
-                        + ":" + exposedPorts.get(0))
-                .with(AUTO_DISCOVER_MEMBERS, false);
+                .with(CONNECTION_STRING, "mongodb://" + mongoDbContainer.getContainerInfo().getConfig().getHostName()
+                        + ":" + exposedPorts.get(0));
```
Comment (rk3rn3r) [inline debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/ConnectorConfiguration.java:75]: Does this return the port inside the container or the outside port? We should make sure we use the right port here. That might have changed as before inside and outside port were equal and not automatically mapped.

Comment (ani-sha) [reply debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/ConnectorConfiguration.java:75]: I believe the inside port is returned here.

**Comment by jpechane** [comment]
@jcechace Could you please review and merge this one? Thanks

File: debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/ConnectorConfiguration.java (hunk 1 above)
Comment (jcechace) [reply debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/ConnectorConfiguration.java:75]: I see this also switches to our contaienr. Apology for not noticing

**Comment by jcechace** [comment]
@ani-sha Depending on what you need to do there are two options

1) Use `MongoDbReplicaSet` instead -- this is a class representing Mongo RS and technically it is the minimal viable unit intended to be run. This class implements `MongoDbDeployment` which directly provides `getConnectionString()` method. 

https://github.com/debezium/debezium/pull/4430# would have to be extended to allow specifying the image when creating `MongoDbReplicaSet` or `MongoDbShardedCluster` (for completeness sake).

2) If you really need to access `MongoDbContainer` directly then I would say we should have the container implement `MongoDbDeployment`

File: debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/ConnectorConfiguration.java (hunk 1 above)
Comment (jcechace) [reply debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/ConnectorConfiguration.java:75]: @ani-sha  also this is not the correct way to get the address here.

If you need an address which is addressable from client use `MongoDbContainer#getClientAddress()`. If ( as seems to be case here) you only need an address which is reachable within docker network, use `MongoDbContaienr#getNamedAddress()`

**Comment by jcechace** [comment]
> @ani-sha Depending on what you need to do there are two options
> 
> 1. Use `MongoDbReplicaSet` instead -- this is a class representing Mongo RS and technically it is the minimal viable unit intended to be run. This class implements `MongoDbDeployment` which directly provides `getConnectionString()` method. This would be preferable
> 
> #4430 would have to be extended to allow specifying the image when creating `MongoDbReplicaSet` or `MongoDbShardedCluster` (for completeness sake).
> 
> 2. If you really need to access `MongoDbContainer` directly then I would say we should have the container implement `MongoDbDeployment`

Hmm thinking about it... If you need an internal connectionString than this interface won't be helpful (as that is intended to provide client connectionString).  So I would say for now, just use the `MongoDbContaienr#getNamedAddress()` as mentioned above.

**Comment by rk3rn3r** [comment]
My 2cents: I agree we should use `MongoDbReplicaSet` in the UI and enable the functionality there too, if this is not inherited down.

File: debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/ConnectorConfiguration.java [hunk 2]
```diff
@@ -68,14 +66,10 @@ public static ConnectorConfiguration forJdbcContainer(JdbcDatabaseContainer<?> j
...
-    public static ConnectorConfiguration forMongoDbContainer(MongoDBContainer mongoDbContainer) {
-        final List<Integer> exposedPorts = mongoDbContainer.getExposedPorts();
-
+    public static ConnectorConfiguration forMongoDbContainer(MongoDbContainer mongoDbContainer) {
         ConnectorConfiguration configuration = new ConnectorConfiguration();
         configuration.with(CONNECTOR, "io.debezium.connector.mongodb.MongoDbConnector")
-                .with(HOSTS, "rs0/" + mongoDbContainer.getContainerInfo().getConfig().getHostName()
-                        + ":" + exposedPorts.get(0))
-                .with(AUTO_DISCOVER_MEMBERS, false);
+                .with(CONNECTION_STRING, mongoDbContainer.getNamedAddress().toString());
```
Comment (jcechace) [inline debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/ConnectorConfiguration.java:72]: this returns a `MongoDbContainer.Address` instance and the `toString()` call resolves to `"<host>:<port>"`, which is not a valid connection string.

**Pull request merged**